### PR TITLE
update dependencies and implement on/off toggle

### LINKED ISSUE #10: update electron, react and other packages
old versions of Electron, React and other packages are used. They should be updated.

### LINKED ISSUE #11: on/off switch for automatically sending the currently played song
It would be great to have an on/off toggle that would allow to choose whether to automatically send the currently played song to Slack or not.
When automatic sending is turned off, a button should be shown allowing manual sending of the currently playing song.

### PR DESCRIPTION
**Pull request by tkothe**
- bump version to 1.1.1
- update npm start command to run `electron .`
- remove electron-template-menu dependency
- add toggle to turn automatic sending on/off (fix #11)

update packages (fix #10)
- update target electron version to 1.26
- update target electron packager version to 7.2.0
- update standard to 7.1.2
- update react version to 0.14.8
- update menubar to 4.1.2
- fix correct reference of menu item when packaging app through `npm run build`

### CODE REVIEW

**Comment by uiur** [comment]
Note: Here is a screenshot for me

image

File: app.css [hunk 1]
```diff
@@ -13,3 +13,9 @@
   right: 15px;
   bottom: 15px;
 }
+
+.btn-sendnow {
+	position: absolute;
+	left:; 15px;
```
Comment (uiur) [inline app.css]: typo? `left:;`

File: app.js [hunk 2]
```diff
@@ -67,25 +79,34 @@ const App = React.createClass({
...
+          React.DOM.input({
+            className: 'checkbox',
+            type: 'checkbox',
+            defaultChecked: self.state.autoSend,
+            onChange: function (e) {
+              self.setState({ autoSend: e.target.checked })
+              window.localStorage.setItem('autoSend', e.target.checked)
+            }
+          }),
+          React.DOM.label({}, 'Send automatically'),
```
Comment (uiur) [inline app.js]: This label should be clickable. Add `htmlFor` attribute to the label.

``` js
React.DOM.input({
  className: 'checkbox',
  id: 'autosend-checkbox',
  ...
}), 
React.DOM.label({ htmlFor: 'autosend-checkbox' }, 'Send automatically'),
```

File: app.js [hunk 3]
```diff
@@ -67,25 +79,34 @@ const App = React.createClass({
               window.localStorage.setItem('listenerName', e.target.value)
             }
           }),
+          React.DOM.input({
+            className: 'checkbox',
+            type: 'checkbox',
+            defaultChecked: self.state.autoSend,
```
Comment (uiur) [inline app.js:86]: I think autoSend should be enabled by default. How do you think?

**Comment by uiur** [comment]
Hi @tkothe. Thanks for your great PR.

I've added a few comments. 
btw, it works for Apple Music now, doesn't it? I'm pretty excited.

**Comment by tkothe** [comment]
@uiureo thanks for the review! I'll do my best to find the time to update based on your comments very soon! (might be a week or two though).

I don't have Apple Music currently (canceled my subscription again recently) but it would indeed be awesome if this worked now!

File: app.js (hunk 3 above)
Comment (tkothe) [inline app.js:86]: it defaults to the last state that was set which is stored in localStorage -> the setting is persisted and picked up again on the next launch.
If nothing is stored in localStorage, it defaults to `false`. See app.js line 15:
https://github.com/tkothe/playing/blob/6dad30b2d5333fa7a80181c6560f71d52273145d/app.js#L15

So if we want to default this to `true`, that's where it should be changed imho.
What do you think?

Comment (uiur) [inline app.js:86]: I prefer to defaurt it `true`. Operations should be minimal at initial setup.

**Comment by tkothe** [comment]
@uiureo, I updated the PR based on your recommendations.

Admittedly, the UI could be nicer showing the label next to the checkbox, but I was unable to get that done in short time - I'm in no way a CSS expert.

**Comment by uiur** [comment]
Great work! Thanks @tkothe. 

I have a plan to update the UI, so no problem.

**Comment by snipe** [comment]
Hi! I know this issue is a billion years old, but I was wondering if this made it into a release? This is a neat app, and I'd love to use the version that has the on/off toggle, but I'm not seeing that from the version I installed from the releases page. 

Is there anything I can help with?

**Comment by uiur** [comment]
Hi @snipe! Good catch. I've forgot that for a billion years. 

I've created the release: https://github.com/uiureo/playing/releases/tag/v1.1.1

Let me know if there's any problem. Thanks!

**Comment by snipe** [comment]
Thank you! Are you still interested in developing this at all? I mean, it does what it says it does, so maybe there’s nothing further to do? 

I’m here because you managed to not only solve my iTunes -> slack problem, but your code is pretty great, and I learned a lot about electron apps while pulling your commits from master into a local app so I could compile/use them.

Are you interested in working on this anymore? I’m a backend dev myself, so React/Electron isn’t my jam specifically, but your project inspired me to install XQuartz and wine, so let me know :)

**Comment by uiur** [comment]
> Are you still interested in developing this at all? 

No. I haven't used this app now, and don't have any plan to work on this. 

But if I had enough time, I would work on:

- improving UI design 
- windows and linux support
- form validation
- updating React / Electron (probably use JSX)

**Comment by snipe** [comment]
I completely understand.

I’ve forked your repo, and if you don’t mind, I may try my hand at tweaking it a bit. I’ll still leave original credits to you of course. It’s a good excuse for me to get my feet wet with Electron, which I’ve been meaning to do forever.

PS - trying to compile this (or anything else) for windows on a Mac is just as awful as you’d imagine.

**Pull request merged**